### PR TITLE
Crdcdh 657

### PR DESCRIPTION
**Pull request by knockknockyoo**

### CODE REVIEW

**Comment by knockknockyoo** [comment]
We might just reuse listBatch API for this. Just spread out the fileInfo array and sort it by the field we need. Let me know if you have any question. Thanks.

For example,
$match : {submissionID: xxx},
$unwind: {
  path: "$files",
},
$sort: {
  "files.fileName": -1, // -1 desc, 1 asc
}

File: services/data-record-service.js [hunk 1]
```diff
@@ -120,6 +120,48 @@ class DataRecordService {
             results:qcResults.slice(offset, offset+first)
         };
     }
+    
+    async listBatchFiles(submissionID, batchID, first, offset, orderBy, sortDirection) {
+        const latestBatch = (await this.batchCollection.find(batchID)).pop();
```
[the remark this answers is not in the document]
Comment (Alejandro-Vega) [reply services/data-record-service.js:127]: @knockknockyoo I tested and it is returning an array. In the `MongoDBCollection` class the `find` function converts it to an array.

File: services/submission.js [hunk 2]
```diff
@@ -331,6 +331,30 @@ class Submission {
...
+                ROLES.ADMIN, ROLES.FEDERAL_LEAD, ROLES.CURATOR, // can see submission details for all submissions
+                ROLES.ORG_OWNER, // can see submission details for submissions associated with his/her own organization
+                ROLES.SUBMITTER, // can see submission details for his/her own submissions
+                ROLES.DC_POC // can see submission details for submissions associated with his/her Data Commons
+            ]);
+        const userRole = context.userInfo?.role;
+        let submission = null;
+        if ([ROLES.ORG_OWNER, ROLES.SUBMITTER, ROLES.DC_POC].includes(userRole)){
+            submission = (await this.submissionCollection.find(params.submissionID)).pop();
+        }
```
[the remark this answers is not in the document]
Comment (Alejandro-Vega) [reply services/submission.js:352]: @knockknockyoo I tested and it is returning an array.

**Comment by Alejandro-Vega** [comment]
> We might just reuse listBatch API for this. Just spread out the fileInfo array and sort it by the field we need. Let me know if you have any question. Thanks.
> 
> For example, $match : {submissionID: xxx}, $unwind: { path: "$files", }, $sort: { "files.fileName": -1, // -1 desc, 1 asc }

We would still need to allow sorting by `nodeType` from the DataRecord. I don't think the listBatch API can provide this. Please let me know if I am misunderstanding your suggestion.

**Comment by knockknockyoo** [comment]
Alex, you're right. The filelistBatch API is a completely different API.

File: services/submission.js [hunk 3]
```diff
@@ -364,6 +364,30 @@ class Submission {
...
+            .verifyRole([
+                ROLES.ADMIN, ROLES.FEDERAL_LEAD, ROLES.CURATOR, // can see submission details for all submissions
+                ROLES.ORG_OWNER, // can see submission details for submissions associated with his/her own organization
+                ROLES.SUBMITTER, // can see submission details for his/her own submissions
+                ROLES.DC_POC // can see submission details for submissions associated with his/her Data Commons
+            ]);
+        const userRole = context.userInfo?.role;
+        let submission = null;
+        if ([ROLES.ORG_OWNER, ROLES.SUBMITTER, ROLES.DC_POC].includes(userRole)){
+            submission = (await this.submissionCollection.find(params.submissionID)).pop();
```
Comment (knockknockyoo) [inline services/submission.js:380]: no need to assign submission.

Comment (Alejandro-Vega) [reply services/submission.js]: Updated to use `verifyBatchPermission` function, but I believe we do need the submission to verify the permissions for a user to view a Batch for specific roles

**Review by AustinSMueller — CHANGES_REQUESTED** [review]
This PR will need to be updated following the data model change to satisfy requirements of the user story

**Pull request closed without merging**